lib/views/pages/transaction_summary.dart

```diff
@@ -3,24 +3,77 @@ import 'package:intl/intl.dart';
 import 'package:sun_flutter_capstone/consts/global_style.dart';
 import 'package:sun_flutter_capstone/views/widgets/cards/elevated_card.dart';
 import 'package:sun_flutter_capstone/consts/consts.dart';
+import 'package:sun_flutter_capstone/controllers/transactions_controller.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 
-class TransactionSummary extends StatelessWidget {
-  final double totalBalance;
-  final double totalIncome;
-  final double totalExpenses;
+class TransactionSummary extends StatefulHookConsumerWidget {
   final String currency;
 
   const TransactionSummary({
     Key? key,
-    required this.totalBalance,
-    required this.totalIncome,
-    required this.totalExpenses,
     required this.currency,
   }) : super(key: key);
 
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() =>
+      _TransactionSummary();
+}
+
+class _TransactionSummary extends ConsumerState<TransactionSummary> {
+
   @override
   Widget build(BuildContext context) {
     final amountFormat = AmountFormat();
+    final transactionState = ref.watch(transactionsNotifierProvider);
+    final expensesState = ref.watch(expenseTransactionsProvider);
+    final incomesState = ref.watch(incomeTransactionsProvider);
+
+    getData(data, section){
+      double _totalExpense = 0;
+      double _totalIncome = 0;
+      for (var dt in data) {
```

Comment:
```suggestion
      for (var transaction in data) {
```

suggesting to change for readability?